{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "aff08ce5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/javascript": "IPython.notebook.set_autosave_interval(180000)"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Autosaving every 180 seconds\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/cat/.conda/envs/bmitools/lib/python3.8/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Importing the dtw module. When using in academic works please cite:\n",
      "  T. Giorgino. Computing and Visualizing Dynamic Time Warping Alignments in R: The dtw Package.\n",
      "  J. Stat. Soft., doi:10.18637/jss.v031.i07.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import matplotlib\n",
    "\n",
    "%matplotlib tk\n",
    "%autosave 180\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "\n",
    "import numpy as np\n",
    "#%matplotlib inline\n",
    "\n",
    "import os\n",
    "from utils import *\n",
    "\n",
    "import os\n",
    "\n",
    "import networkx as nx\n",
    "\n",
    "from utils_calcium import (ProcessCalcium, \n",
    "                           get_reward_triggered_psth, \n",
    "                           plot_multi_session_psth_imshow, \n",
    "                           plot_psth,\n",
    "                             compute_ensemble_only_correlations)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "044f5ae6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/9 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " ***** USING non-merged binarization\n",
      "remove bad cells:  True\n",
      "...setting 2p parameters...\n",
      "...loading binarized traces from file:  /media/cat/8TB/donato/bmi/DON-015821/20230519/plane0/binarized_traces.npz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 11%|█         | 1/9 [00:02<00:16,  2.08s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "couldn't reward results.npz file...\n",
      " ***** USING non-merged binarization\n",
      "remove bad cells:  True\n",
      "...setting 2p parameters...\n",
      "...loading binarized traces from file:  /media/cat/8TB/donato/bmi/DON-015821/20230520/plane0/binarized_traces.npz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 22%|██▏       | 2/9 [00:09<00:34,  5.00s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "couldn't reward results.npz file...\n",
      " ***** USING non-merged binarization\n",
      "remove bad cells:  True\n",
      "...setting 2p parameters...\n",
      "...loading binarized traces from file:  /media/cat/8TB/donato/bmi/DON-015821/20230521/plane0/binarized_traces.npz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 33%|███▎      | 3/9 [00:13<00:29,  4.83s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "couldn't reward results.npz file...\n",
      " ***** USING non-merged binarization\n",
      "remove bad cells:  True\n",
      "...setting 2p parameters...\n",
      "...loading binarized traces from file:  /media/cat/8TB/donato/bmi/DON-015821/20230522/plane0/binarized_traces.npz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 44%|████▍     | 4/9 [00:24<00:35,  7.09s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "couldn't reward results.npz file...\n",
      " ***** USING non-merged binarization\n",
      "remove bad cells:  True\n",
      "...setting 2p parameters...\n",
      "...loading binarized traces from file:  /media/cat/8TB/donato/bmi/DON-015821/20230523/plane0/binarized_traces.npz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 56%|█████▌    | 5/9 [00:35<00:34,  8.62s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "couldn't reward results.npz file...\n",
      " ***** USING non-merged binarization\n",
      "remove bad cells:  True\n",
      "...setting 2p parameters...\n",
      "...loading binarized traces from file:  /media/cat/8TB/donato/bmi/DON-015821/20230524/plane0/binarized_traces.npz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 67%|██████▋   | 6/9 [00:44<00:26,  8.76s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "couldn't reward results.npz file...\n",
      " ***** USING non-merged binarization\n",
      "remove bad cells:  True\n",
      "...setting 2p parameters...\n",
      "...loading binarized traces from file:  /media/cat/8TB/donato/bmi/DON-015821/20230525/plane0/binarized_traces.npz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 78%|███████▊  | 7/9 [00:56<00:19,  9.61s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "couldn't reward results.npz file...\n",
      " ***** USING non-merged binarization\n",
      "remove bad cells:  True\n",
      "...setting 2p parameters...\n",
      "...loading binarized traces from file:  /media/cat/8TB/donato/bmi/DON-015821/20230526/plane0/binarized_traces.npz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 89%|████████▉ | 8/9 [01:08<00:10, 10.40s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "couldn't reward results.npz file...\n",
      " ***** USING non-merged binarization\n",
      "remove bad cells:  True\n",
      "...setting 2p parameters...\n",
      "...loading binarized traces from file:  /media/cat/8TB/donato/bmi/DON-015821/20230527/plane0/binarized_traces.npz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 9/9 [01:20<00:00,  8.94s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "session_ids:  [0 1 2 3 4 5 6 7 8]\n",
      ".... already computed...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ".... already computed...\n",
      ".... already computed...\n",
      ".... already computed...\n",
      ".... already computed...\n",
      ".npz corrupt\n",
      ".... already computed...\n",
      ".... already computed...\n",
      ".... already computed...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "computing ensemble only correlations:  67%|██████▋   | 6/9 [00:04<00:02,  1.29it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "missing file:  /media/cat/8TB/donato/bmi/DON-015821/20230524/ensembles/ensembles_matches_bmi_to_suite2p.npz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "computing ensemble only correlations: 100%|██████████| 9/9 [00:07<00:00,  1.28it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... done processing:  DON-015821\n",
      " \n",
      " \n",
      "....DONE...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "####################################################\n",
    "################## LOAD DATA #######################\n",
    "####################################################\n",
    "\n",
    "root_dir = '/media/cat/8TB/donato/bmi/'\n",
    "animal_ids = [\n",
    "    # M1 mice\n",
    "    #\"DON-011733\",    # M1    - Processed\n",
    "    #\"DON-014618\",    # M1    - Processed\n",
    "    #'DON-014451',    # M1    - Processed\n",
    "    #'DON-014382',    # M1    - Processed\n",
    "    #'DON-014575',    # M1    - Processed\n",
    "    #'DON-014618',    # M1    - Processed\n",
    "    #'DON-015821',    # M1    - Processed\n",
    "    # 'DON-017923',   # M1    - Not Processed <has extra days at the end....\n",
    "    # 'DON-018129',   # M1    - Not Processed\n",
    "    #'DON-018130',   # M1    - Not Processed\n",
    "\n",
    "    # MAYBE MICE\n",
    "    # 'DON-013392'  <-- only last day was good, all others were at chance\n",
    "    # 'DON-015962'  <-- might be ok, rechedk it.\n",
    "\n",
    "    # CA3 mice\n",
    "    #\"DON-012502\",    # CA3    - spreadsheet is old - need reprocessing; ALSO Weird performance, may exclude\n",
    "    #\"DON-014266\",    # CA3   - Processed\n",
    "    #'DON-014371',    # CA3   - Processed\n",
    "    'DON-015801',    # CA3   - Processed\n",
    "    #'DON-016658',    # CA3   - Processed\n",
    "    #\"DON-019683\",    # CA3   - Processed\n",
    "]\n",
    "#\n",
    "#for animal_id in animal_ids:\n",
    "#animal_id = animal_ids[0]\n",
    "animal_id = animal_ids[0]\n",
    "\n",
    "for animal_id in animal_ids:\n",
    "\n",
    "    #       \n",
    "    c = ProcessCalcium(root_dir,\n",
    "                    animal_id)\n",
    "\n",
    "    # \n",
    "    c.remove_bad_cells = False\n",
    "    c.use_non_merged = True\n",
    "\n",
    "    #\n",
    "    c.recompute_binarization = False  # IF TRue: must input correct params from bin notebooks\n",
    "    c.dff_min = 0.05                  # min %DFF for [ca] burst to considered a spike (default 5%) overwrites percentile threshold parameter\n",
    "    c.percentile_threshold = 0.999   # this is pretty fixed, we don't change it; we want [ca] bursts that are well outside the \"physics-caused\"noise\n",
    "    c.maximum_std_of_signal = 0.25     # if std of signal is greater than this, then we have a noisy signal and we don't want to binarize it\n",
    "\n",
    "    #\n",
    "    c.save_figures = c.recompute_binarization\n",
    "    c.load_sessions()\n",
    "\n",
    "    #\n",
    "    if True:\n",
    "        c.compute_bmi_to_suite2p_matches()\n",
    "        compute_ensemble_only_correlations(c)\n",
    "\n",
    "    print (\"... done processing: \", animal_id)\n",
    "    print (\" \")\n",
    "    print (\" \")\n",
    "#\n",
    "print (\"....DONE...\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "27e81865",
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "'numpy.int64' object is not iterable",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[3], line 12\u001b[0m\n\u001b[1;32m      9\u001b[0m c\u001b[38;5;241m.\u001b[39muse_DFF \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[1;32m     11\u001b[0m \u001b[38;5;66;03m#\u001b[39;00m\n\u001b[0;32m---> 12\u001b[0m \u001b[43mc\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mvisualize_ensemble_to_reward_average\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     14\u001b[0m \u001b[38;5;66;03m#\u001b[39;00m\n",
      "File \u001b[0;32m~/code/bmitools/bmitools/analysis/calcium/ensembles/utils_calcium.py:4524\u001b[0m, in \u001b[0;36mProcessCalcium.visualize_ensemble_to_reward_average\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   4522\u001b[0m  \u001b[38;5;66;03m#\u001b[39;00m\n\u001b[1;32m   4523\u001b[0m  \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[0;32m-> 4524\u001b[0m      psths_avg, psths_shuffled_avg, n_bursts  \u001b[38;5;241m=\u001b[39m \u001b[43mget_reward_triggered_bmi_ensembles\u001b[49m\u001b[43m(\u001b[49m\u001b[43msession_id\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m   4525\u001b[0m \u001b[43m     \u001b[49m\n\u001b[1;32m   4526\u001b[0m \u001b[43m                                                                               \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m   4527\u001b[0m \u001b[43m                                                                               \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwindow\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   4528\u001b[0m \u001b[38;5;66;03m# except:\u001b[39;00m\n\u001b[1;32m   4529\u001b[0m      \u001b[38;5;66;03m# print (\"error in session: \", session_id)\u001b[39;00m\n\u001b[1;32m   4530\u001b[0m      \u001b[38;5;66;03m# continue\u001b[39;00m\n\u001b[1;32m   4531\u001b[0m  \n\u001b[1;32m   4532\u001b[0m  \u001b[38;5;66;03m# \u001b[39;00m\n\u001b[1;32m   4533\u001b[0m  \u001b[38;5;28;01mfor\u001b[39;00m k \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;241m4\u001b[39m):\n",
      "File \u001b[0;32m~/code/bmitools/bmitools/analysis/calcium/ensembles/utils_calcium.py:5231\u001b[0m, in \u001b[0;36mget_reward_triggered_bmi_ensembles\u001b[0;34m(session_id, c, window, n_tests)\u001b[0m\n\u001b[1;32m   5228\u001b[0m n_bursts\u001b[38;5;241m.\u001b[39mappend(idx\u001b[38;5;241m.\u001b[39mshape[\u001b[38;5;241m0\u001b[39m])\n\u001b[1;32m   5230\u001b[0m \u001b[38;5;66;03m#\u001b[39;00m\n\u001b[0;32m-> 5231\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m r \u001b[38;5;129;01min\u001b[39;00m reward_times:\n\u001b[1;32m   5232\u001b[0m     temp2 \u001b[38;5;241m=\u001b[39m temp[r\u001b[38;5;241m-\u001b[39mwindow:r\u001b[38;5;241m+\u001b[39mwindow]\n\u001b[1;32m   5233\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m temp2\u001b[38;5;241m.\u001b[39mshape[\u001b[38;5;241m0\u001b[39m]\u001b[38;5;241m==\u001b[39mwindow\u001b[38;5;241m*\u001b[39m\u001b[38;5;241m2\u001b[39m:\n",
      "\u001b[0;31mTypeError\u001b[0m: 'numpy.int64' object is not iterable"
     ]
    }
   ],
   "source": [
    "##################################################################\n",
    "###### visualize the reward response of each ROI over time #######\n",
    "##################################################################\n",
    "\n",
    "# \n",
    "c.window = 300\n",
    "c.show_random = True\n",
    "c.smoothing = False\n",
    "c.use_DFF = False\n",
    "\n",
    "#\n",
    "c.visualize_ensemble_to_reward_average()\n",
    "\n",
    "#\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "1ac4ef21",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 70/70 [00:06<00:00, 11.02it/s]      | 0/8 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[8, 0, -0.020935156387119958, 0.2516654622719943, -0.969609836112199], [8, 1, 0.009465419524277098, 0.6042922748801083, 0.46360811590115064], [8, 2, 0.0017234704804849695, 0.9248239075922887, 0.11642169569290028], [8, 3, 0.08517564349945184, 2.9855780028638557e-06, 4.4075178203680885], [8, 4, -0.013240072147209767, 0.4685022332320996, -0.6999581728271363], [8, 5, 0.014136267204821466, 0.43893597340262897, 0.615366884149057], [8, 6, -0.00461685067113489, 0.8004449563251972, -0.2563648443705459], [8, 7, -0.019807777138684323, 0.27811175043112163, -1.0022863154561883], [8, 8, 0.6392628021807613, 0.0, 23.162322501686436], [8, 9, -0.017962080440940442, 0.3253648608045053, -0.904818631104189], [8, 10, 0.00272514922824489, 0.8813953127572652, 0.1371854138541682], [8, 11, 0.0321733090816712, 0.07808172958194311, 1.7516740152113333], [8, 12, 0.03644121399871841, 0.0459558555992091, 1.761182407183843], [8, 13, -0.015129660606574434, 0.40745097088200494, -0.7504193363603234], [8, 14, -0.018723593675987037, 0.3052707449587022, -0.9147644283797206], [8, 15, 0.017964958991237706, 0.3252873205121187, 0.8642421411572977], [8, 16, 0.0362151812039601, 0.0473215269967042, 1.6141647199941298], [8, 17, -0.024494945683784342, 0.1798277631707405, -1.1889072195080717], [8, 18, 0.027708127090871892, 0.12919210669668363, 1.3323997375052656], [8, 19, 0.037152259169052966, 0.041874278432835525, 1.754336250965993], [8, 20, -0.009203631251926123, 0.614328675889822, -0.49021771957758037], [8, 21, -0.008373156620651055, 0.6466409678408648, -0.40080473911612224], [8, 22, -0.005736095002196248, 0.7534826743133591, -0.2856713789672173], [8, 23, 0.0027992009236085425, 0.8781972223496246, 0.19341750038566263], [8, 24, -0.0041796444841943415, 0.8189978224004196, -0.1910186858579553], [8, 25, -0.016172681088353555, 0.3758854169577912, -0.8005684244333983], [8, 26, 0.009494522446996207, 0.6031810890473832, 0.49365554637258635], [8, 27, -0.00444674921766017, 0.8076504728410941, -0.1786321250983702], [8, 28, 5.759670754706875e-05, 0.997483965457873, 0.0003541297782556335], [8, 29, 0.04088012072387166, 0.025149458353811816, 2.0389157699884746], [8, 30, -0.01420468058377516, 0.43672384220379923, -0.7194315975993809], [8, 31, 0.0014356067945348154, 0.9373518266568923, 0.05144558980439758], [8, 32, -0.015520209230073622, 0.3954508696078594, -0.6918181291004878], [8, 33, 0.005282727918627082, 0.7724066320267654, 0.23843355726707385], [8, 34, 0.012067072917793006, 0.5088115150774926, 0.5812085367979145], [8, 35, -0.019081571799976608, 0.2961152811641013, -0.888658698350396], [8, 36, -0.02711699211945896, 0.1375669002400376, -1.3330300388645058], [8, 37, -0.020589944435123703, 0.25957034727765654, -1.0373191027046185], [8, 38, 0.022753598103632408, 0.21279824390097485, 1.13787406536835], [8, 39, -0.03334396239565637, 0.06783961811242122, -1.6338890161699269], [8, 40, 0.023904734894463247, 0.19054870235513535, 1.185471109492897], [8, 41, -0.002321309578497998, 0.8988687335399473, -0.13941868925866788], [8, 42, -0.02141869862550531, 0.2408775975333833, -1.09109604476334], [8, 43, -0.01359558489840747, 0.45664363996439944, -0.6674330330529816], [8, 44, 0.05634298766576088, 0.00202052364775717, 2.902190697064119], [8, 45, 0.0742124659203221, 4.727587804851466e-05, 3.5682152844928297], [8, 46, 0.012592629270397011, 0.4905296818119106, 0.6820822567270829], [8, 47, 0.06516754761626696, 0.00035469466001135, 3.230954902168979], [8, 48, -0.005504156078917807, 0.7631463300878426, -0.22604101897646478], [8, 49, -0.005024889766837978, 0.783230511175738, -0.27635375216602426], [8, 50, -0.009725293297209595, 0.5944026943486568, -0.5044763205790284], [8, 51, -3.504607643005286e-05, 0.9984690577881281, -0.024455267757614328], [8, 52, -0.009998514060554483, 0.5840856682341903, -0.5148359876511331], [8, 53, 0.0072733878989465805, 0.6904686838354963, 0.3752578090878552], [8, 54, 0.03348678305617333, 0.06666934872187232, 1.7920931183660929], [8, 55, 0.008239792093588725, 0.6518947219836215, 0.3539558785925164], [8, 56, -0.01960425258294576, 0.2830805633970938, -0.9874402675083602], [8, 57, 0.03618444838318533, 0.04750982027783362, 1.9349026896182626], [8, 58, 0.00013353733734493465, 0.9941666436742344, 0.0058611250844748984], [8, 59, -0.020903048639143766, 0.2523935281057139, -0.9954295260466108], [8, 60, -0.02344160939226257, 0.19928506104712856, -1.2444789573947252], [8, 61, -0.000570263235809536, 0.9750928224372325, -0.04872727395066513], [8, 62, 0.00071103396074713, 0.9689472276801093, -0.021206075264131107], [8, 63, -0.02257232239828556, 0.21646628057872003, -1.1036384836849953], [8, 64, -0.0056200815615555216, 0.7583116169396925, -0.3080798994152053], [8, 65, 0.011320885793190983, 0.5353684637919611, 0.5231713486103297], [8, 66, 0.022530393634015326, 0.21732113015752721, 1.229540454210901], [8, 67, -0.0019362833265793046, 0.9155740157498644, -0.1314261475990678], [8, 68, -0.014651612522095934, 0.4224312123943472, -0.778820235401368], [8, 69, -0.0031514553606261956, 0.8630123792384351, -0.15396380706228463]]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "100%|██████████| 70/70 [00:06<00:00, 10.89it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[15, 0, 0.015545120777868076, 0.39469274297135337, 0.6503553777086702], [15, 1, -0.0025414223465364456, 0.8893382175405914, -0.10747798468007565], [15, 2, -0.0018934635300952233, 0.9174342972785362, -0.10798759137716973], [15, 3, 0.14785547470451105, 3.960046946231817e-16, 6.774974460315877], [15, 4, -0.016664586464393342, 0.3615371628304066, -0.7484749845541881], [15, 5, -0.011295339634000952, 0.536289865315776, -0.501599906911766], [15, 6, 0.04834059233045648, 0.008092929071669168, 2.301686752124089], [15, 7, -0.006372354428011435, 0.7271746616653495, -0.3099031237583239], [15, 8, -0.006889227297901271, 0.7060355764193234, -0.3729947018920602], [15, 9, 0.0017703700102031632, 0.9227845064730642, 0.15081017772692246], [15, 10, 0.0899123477119391, 8.112443285498823e-07, 4.2621562436162055], [15, 11, 0.06559660140379664, 0.00032408146360465525, 3.3217383272397005], [15, 12, -0.020302870957442397, 0.26627353308479756, -0.8504344022967764], [15, 13, -0.007019698039031165, 0.7007344695200349, -0.32805862227810834], [15, 14, -0.015015503829410495, 0.4109991485569695, -0.7526435476606993], [15, 15, 0.6552345051504243, 0.0, 18.528376725793468], [15, 16, -0.01099057797535015, 0.5473431931557247, -0.44080948344282167], [15, 17, 0.03589779054890674, 0.049296624966799, 1.6843503865921323], [15, 18, -0.017821799478322357, 0.32915820765069187, -0.8241868431443277], [15, 19, -0.003504101901470758, 0.8478611380652784, -0.17890210537972695], [15, 20, 0.06193465055873145, 0.0006884312284183097, 3.110850306553591], [15, 21, -0.0036144621878604006, 0.8431308235964438, -0.1755099923670697], [15, 22, 0.0014332022630808936, 0.9374565422980831, 0.08028814274127304], [15, 23, 0.010076777989976315, 0.5811457875166025, 0.46433402636637605], [15, 24, -0.016601523097764644, 0.36335722104213264, -0.7728529747529677], [15, 25, 0.00818604136005813, 0.6540170865725785, 0.44161512283779375], [15, 26, -0.0053372873249832335, 0.7701218476918691, -0.265491195821663], [15, 27, -0.007944375285801985, 0.6635937276334227, -0.37028715332664375], [15, 28, -0.0072401430421511655, 0.6918107798090961, -0.29440167830683667], [15, 29, -0.009530903361564516, 0.6017933139715088, -0.4634754135294786], [15, 30, -0.004294393563769257, 0.8141181711164766, -0.1893283760008311], [15, 31, -0.006902704240896047, 0.7054873338815191, -0.33100856588566124], [15, 32, -0.021075457286877687, 0.24850122282390202, -0.8723729095445985], [15, 33, 0.012328153819187043, 0.49968527078962405, 0.5875362182358688], [15, 34, -0.012177474722328444, 0.5049417283925184, -0.6235849746661238], [15, 35, -0.01972920150475939, 0.28002300432448163, -0.8839258078785449], [15, 36, -0.02708940590537131, 0.13796770786503162, -1.3126761385243466], [15, 37, -0.002830786225410461, 0.8768337405228687, -0.09622832690950282], [15, 38, 0.009671436902174483, 0.5964461355968506, 0.44396067983251575], [15, 39, -0.010267505940942823, 0.5740104791169741, -0.4947343129800503], [15, 40, 0.013967761195034891, 0.44441193197220635, 0.6027160927087191], [15, 41, -0.012239909652244531, 0.5027601510703106, -0.6125313986013592], [15, 42, 0.0045581724072059245, 0.8029286905141839, 0.2547938808963402], [15, 43, -0.022078757424532594, 0.2266833107362946, -1.0666986728201795], [15, 44, 0.027739276446985237, 0.12876203222740862, 1.4516482450955108], [15, 45, 0.05579779763565168, 0.0022335593783241923, 2.6536547019144847], [15, 46, -0.009363519634205777, 0.6081901565810106, -0.4758111795751711], [15, 47, 0.12150003450720118, 2.4420519846496204e-11, 5.828270249226575], [15, 48, -1.8791080177460147e-05, 0.9991791357734757, 0.005210731481918135], [15, 49, -0.00782816399713982, 0.6682186544665192, -0.35680627447933944], [15, 50, -0.016657114023341742, 0.36175252451558076, -0.7772354305795312], [15, 51, 0.004647652866181742, 0.7991419657054968, 0.20101552292398825], [15, 52, -0.009114172992651397, 0.6177750613127322, -0.4446970840451145], [15, 53, -0.012781311009250621, 0.4840534205059745, -0.6163514190403073], [15, 54, -0.006992981538826513, 0.7018188022249695, -0.3637808385561732], [15, 55, 0.0452891603371319, 0.013107738338762118, 2.339540222133322], [15, 56, -0.0053412107679330564, 0.7699576222377934, -0.2668857039815921], [15, 57, -0.009959749567837808, 0.5855443629548761, -0.46499617187397235], [15, 58, 0.0020804170986202745, 0.9093156602776481, 0.05875672162087639], [15, 59, 0.027149901717922496, 0.1370899221860191, 1.2356556308877154], [15, 60, 0.01081783225312527, 0.5536581018607806, 0.591760885942526], [15, 61, -0.008609228169522703, 0.6373843422052649, -0.40308713550873154], [15, 62, -0.002205862886304857, 0.9038733063602018, -0.1431250630996546], [15, 63, -0.0006980308887656475, 0.969514828083595, -0.012610079941703288], [15, 64, 0.0014565760234625962, 0.9364386776932258, 0.10662627094988752], [15, 65, 0.005626592690318339, 0.7580403460166707, 0.31427807715472955], [15, 66, 0.05369576503948763, 0.0032616846063821194, 2.6112742572151326], [15, 67, 0.06659287446557557, 0.0002622670534065554, 3.2075593054742857], [15, 68, 0.005281100339750945, 0.7724748205619496, 0.2743077129325047], [15, 69, -0.0032551415729960434, 0.8585520752202908, -0.20631110942564926]]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "100%|██████████| 70/70 [00:06<00:00, 11.10it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[11, 0, 0.019558860381308385, 0.28419690997697866, 0.9121659797911661], [11, 1, -0.001790329405701157, 0.9219167358701278, -0.024954394688678317], [11, 2, 0.06145710871570109, 0.0007573608890032846, 3.0199466730105593], [11, 3, 0.06978447151383316, 0.00013060021948167343, 3.3640032105383155], [11, 4, -0.01741753106779209, 0.3402495819931238, -0.8191046125294942], [11, 5, 0.0210495783290304, 0.24908277526356062, 1.161686327776866], [11, 6, 0.0071327255878847915, 0.6961538252695101, 0.2946547932252912], [11, 7, 0.008865356953401625, 0.627404849336674, 0.3966832733993264], [11, 8, 0.003957101591262636, 0.8284811921533015, 0.2979537943504044], [11, 9, 0.04682100063358891, 0.010322646332811061, 2.314377467166485], [11, 10, 0.0401831376812459, 0.02774361745121561, 1.7569650613073216], [11, 11, 0.6438826149379797, 0.0, 18.326295047708452], [11, 12, 0.03422182096816427, 0.060906431668237886, 1.494612090175488], [11, 13, 0.021453727504211165, 0.2401089423606445, 0.9777548019599204], [11, 14, -0.003292452599984744, 0.8569481501538597, -0.17805743260341483], [11, 15, -0.011862127281150069, 0.5160362139111129, -0.5565716889085189], [11, 16, -0.0010963438096043105, 0.9521363476734137, -0.046005541997833], [11, 17, -0.015496141608119775, 0.3961841480135568, -0.7398554731496176], [11, 18, -0.01283347946754074, 0.4822710182154011, -0.6169972232140905], [11, 19, 0.025885900133726797, 0.15634387195581279, 1.184796618455168], [11, 20, 0.00039464940870439927, 0.9827615845661033, -0.04176844483212169], [11, 21, -0.008083670613204273, 0.65806695954145, -0.36732865558506606], [11, 22, 0.028020458771333055, 0.12492995109917296, 1.5522446995205164], [11, 23, -0.014322682837553582, 0.4329233889485367, -0.6246110677815292], [11, 24, 0.002152672962235625, 0.9061803678221397, 0.07994930412579394], [11, 25, 0.0011599627876417076, 0.9493625286275619, 0.017167134837974513], [11, 26, 0.0640567130252839, 0.0004469718007044281, 3.2242108928222346], [11, 27, -0.006315272354718363, 0.7295223317849338, -0.31002280295997164], [11, 28, 0.06145200535456965, 0.0007581309576304094, 3.2170603486307385], [11, 29, 0.006699652500789378, 0.7137635309947399, 0.22683369789650062], [11, 30, 0.011185275549430482, 0.5402687681891643, 0.5759811613940594], [11, 31, 0.04704892531314092, 0.00995671090244688, 2.679064340582624], [11, 32, -0.018775967255351533, 0.3039196684252847, -0.8345958767553693], [11, 33, -0.01002445111004753, 0.5831106109452603, -0.511314855274806], [11, 34, -0.009139882590725198, 0.6167837314108711, -0.45910457008373007], [11, 35, -0.016004008068534418, 0.3808852114063874, -0.8112858295485414], [11, 36, -0.013954007863662845, 0.44486058892670344, -0.6709227620103967], [11, 37, 0.04746258045882221, 0.00932198289819997, 2.1445035572578637], [11, 38, 0.04230174945501231, 0.02050189219247102, 2.2339178871766894], [11, 39, -0.015769962279996972, 0.3878900283424641, -0.7822355435627881], [11, 40, 0.17640533086669136, 2.1482062666536376e-22, 7.846769531207289], [11, 41, 0.0035040998125811717, 0.8478612276526609, 0.21148605110506372], [11, 42, -0.02383046391035773, 0.19193044277188512, -1.2003764164425375], [11, 43, -0.0044492132747108795, 0.8075459770006398, -0.21870924308528125], [11, 44, 0.02769216209380892, 0.12941296485216783, 1.3282150095394714], [11, 45, 0.014691241429189456, 0.4211772622086095, 0.7318540640354511], [11, 46, -0.019468913108138815, 0.28641780302993824, -0.947043251346409], [11, 47, 0.06251289909544258, 0.0006127739074229576, 3.1778395092515663], [11, 48, 0.049862843581295405, 0.006301585570744033, 2.563029060709603], [11, 49, -0.006252963083631172, 0.7320878813951609, -0.3175656875315408], [11, 50, -0.0086554376052929, 0.6355789583256971, -0.4150503241232904], [11, 51, -0.01511306955400366, 0.4079655111186216, -0.701795910173987], [11, 52, 0.0043289950259704345, 0.8126481643850679, 0.21453888300454602], [11, 53, 0.02923547780968763, 0.10938402437677801, 1.4255758250411064], [11, 54, 0.05238444721189227, 0.004104858919614134, 2.6244565174670305], [11, 55, 0.3827483687891756, 2.8617494450190716e-105, 15.11499068726003], [11, 56, 0.020638322316335035, 0.25845232202601, 1.0960544677094854], [11, 57, 0.0017938977861386928, 0.9217616038745389, 0.129349259108599], [11, 58, 0.020972278458048575, 0.2508255196136413, 1.0971848365863224], [11, 59, 0.011598462323240443, 0.5254084874457917, 0.49884432620956554], [11, 60, -0.021405181720034316, 0.24117466802318455, -1.051788178752116], [11, 61, 0.0035130593128770374, 0.8474769945702518, 0.2339700296889687], [11, 62, 0.007220253241805346, 0.692614194869734, 0.33193952345794986], [11, 63, 0.0005752448335917255, 0.9748753146051953, 0.056867858295949654], [11, 64, 0.006721244883094864, 0.7128818242711078, 0.40188374857652476], [11, 65, 0.007665238742895866, 0.6747239034383552, 0.39549188586650136], [11, 66, 0.5054873690292006, 2.340238302120494e-194, 15.489182266033378], [11, 67, 0.03749059137935111, 0.04004249117237434, 1.892365670692173], [11, 68, 0.056376645716319555, 0.0020080031727456333, 2.510902649010799], [11, 69, 0.042986936370377114, 0.018542722822099955, 2.0108792803105096]]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "100%|██████████| 70/70 [00:06<00:00, 10.38it/s]\n",
      "computing ensemble only correlations: 100%|██████████| 8/8 [00:29<00:00,  3.70s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[1, 0, -0.03691533545343203, 0.043198811487776254, -1.441813963073793], [1, 1, 0.6491232968452675, 0.0, 18.22373841955697], [1, 2, -0.01640502980491762, 0.36906482705406873, -0.7479152601771379], [1, 3, -0.026560987429834107, 0.1458200635011077, -1.2499827021985945], [1, 4, -0.010169939493655248, 0.5776553480048494, -0.4782610103026933], [1, 5, 0.009969327708187311, 0.5851837835489013, 0.5380977269118613], [1, 6, -0.014440513555881414, 0.42914761746651386, -0.7787960020924011], [1, 7, -0.029167625806573033, 0.11020983871435065, -1.3831699128084696], [1, 8, -0.015985590606803945, 0.381433601584218, -0.7860481234742338], [1, 9, -0.00871485467211703, 0.6332607364954758, -0.362049853073207], [1, 10, -0.015552360719061941, 0.39447257661792207, -0.7653859676409526], [1, 11, 0.008358094152894248, 0.6472334628281192, 0.4027220175481502], [1, 12, -0.002337763396622, 0.8981557888521792, -0.11948851028026852], [1, 13, -0.016121833843241482, 0.377388345922245, -0.8003348832028383], [1, 14, -0.007144136687545077, 0.6956919816763178, -0.3155795325686245], [1, 15, 0.021993867907027433, 0.22847465666528563, 1.03672004884598], [1, 16, 0.014331501412524273, 0.43264014274449764, 0.7497964180458074], [1, 17, -0.019458932956578127, 0.2866649436753206, -0.8614778697349577], [1, 18, 0.02648303109836207, 0.14700688719259894, 1.0870342740852488], [1, 19, 0.0006637482061954158, 0.9710113834259257, -0.05589883126872334], [1, 20, -0.003638192741205032, 0.8421144058479408, -0.21204598308930103], [1, 21, -0.004281360851569352, 0.8146720221620457, -0.16201914203608653], [1, 22, -0.008068607744253187, 0.658663710281939, -0.41238505736600667], [1, 23, -0.02334807067117363, 0.20108452510323607, -1.1040312358828157], [1, 24, -0.024207814890478154, 0.18498604770063823, -1.1699141654931924], [1, 25, -0.014427187405034516, 0.42957367910272287, -0.7245338762103027], [1, 26, 0.017016513325218263, 0.3514854592182724, 0.7587661876727773], [1, 27, -0.015483284489041283, 0.39657620734581084, -0.7516064772885784], [1, 28, -0.004054536117238881, 0.8243259868516208, -0.20315306992526164], [1, 29, -0.0181905755676745, 0.3192472102178507, -0.836200997830709], [1, 30, 0.0023462408378987372, 0.8977884929693314, 0.10233754116319778], [1, 31, 0.01941750509887339, 0.2876923690988881, 1.046135857123842], [1, 32, -0.032190380108799384, 0.07792380476189191, -1.418756470264198], [1, 33, 0.02946580642837624, 0.10661716096703353, 1.3939200377686396], [1, 34, -0.014605656462001994, 0.42388810267123367, -0.7079283365822314], [1, 35, -0.02686356110575523, 0.14128297593813793, -1.0942831700829236], [1, 36, -0.014095982200369008, 0.4402415840491512, -0.7076104580402138], [1, 37, -0.02207438526895562, 0.2267753270632952, -1.0520681198834645], [1, 38, 0.03727752486193309, 0.04118796909933158, 1.8812526235729845], [1, 39, -0.015085829575625972, 0.4088111464676935, -0.7304642297667172], [1, 40, 0.001040768888593162, 0.954559917375309, 0.0794663682229259], [1, 41, 0.010783933144793154, 0.554901494272412, 0.5950902878431971], [1, 42, -0.00363918915960351, 0.8420717333853524, -0.166055448449381], [1, 43, -0.025597707770382765, 0.16100796909050888, -1.2191251028798569], [1, 44, -0.015858152768987305, 0.38524138942657554, -0.8010812605749749], [1, 45, -0.01981666521756982, 0.2778961191362511, -0.9351268308155467], [1, 46, -0.021251630967273647, 0.24456746931729545, -1.1062111372350012], [1, 47, -0.013944333603566787, 0.4451763342986711, -0.673358138062619], [1, 48, 0.007507968475445552, 0.6810264977080418, 0.3398577690464925], [1, 49, -0.016601286013355032, 0.36336407427789785, -0.8221578728308534], [1, 50, -0.017057014294523637, 0.3503401363160462, -0.8214111936199936], [1, 51, -0.027878176133846078, 0.12685778629266264, -1.2803876735145243], [1, 52, -0.014348961495313204, 0.43207965402090737, -0.6736173418460389], [1, 53, -0.017039713107818027, 0.3508291050943708, -0.7579954017572285], [1, 54, -0.01991952532350261, 0.2754089346123867, -0.9332925114474268], [1, 55, 0.061302250242888115, 0.000781054315807795, 3.128503532378628], [1, 56, -0.015467028143210264, 0.3970722561102995, -0.7564026134212779], [1, 57, -0.006574209888638369, 0.7188934011837363, -0.3032601049970369], [1, 58, 0.004227721631090671, 0.8169524995004075, 0.17320938314634324], [1, 59, -0.024011118548263377, 0.18858230915880836, -1.1371884991754508], [1, 60, 0.01906624713797894, 0.2965034135408284, 0.9389136835584432], [1, 61, -0.020998471263603647, 0.25023404608333993, -1.0082050613584734], [1, 62, 0.022857115814954847, 0.21072383793449018, 1.033063615752219], [1, 63, -0.02395295481128476, 0.1896555447556884, -1.1706481659803547], [1, 64, -0.008179664618971755, 0.6542690602668676, -0.41086033484066825], [1, 65, -0.026726168004758702, 0.14332955920997809, -1.3367640117416972], [1, 66, 0.0020091723453487483, 0.9124084621730464, 0.125943651599588], [1, 67, -0.01916482303173633, 0.29401269202447716, -0.877604819966566], [1, 68, -0.005826583389703112, 0.7497228542033746, -0.286586774659376], [1, 69, -0.022441492406832558, 0.2191416826319748, -1.053749548181749]]\n",
      "fname_ensemble_matching:  /media/cat/8TB/donato/bmi/DON-019683/20240120/ensembles/ensembles_matches_bmi_to_suite2p.npz\n",
      "idxs:  [ 8 15 11  1]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2157941/4053755067.py:63: MatplotlibDeprecationWarning: Auto-removal of overlapping axes is deprecated since 3.6 and will be removed two minor releases later; explicitly call ax.remove() as needed.\n",
      "  ax=plt.subplot(2,2,ctr+1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fname_ensemble_matching:  /media/cat/8TB/donato/bmi/DON-019683/20240121/ensembles/ensembles_matches_bmi_to_suite2p.npz\n",
      "idxs:  [17 52 27  1]\n",
      "fname_ensemble_matching:  /media/cat/8TB/donato/bmi/DON-019683/20240122/ensembles/ensembles_matches_bmi_to_suite2p.npz\n",
      "idxs:  [10 23 12 55]\n",
      "fname_ensemble_matching:  /media/cat/8TB/donato/bmi/DON-019683/20240123/ensembles/ensembles_matches_bmi_to_suite2p.npz\n",
      "idxs:  [ 7 29 19  5]\n",
      "fname_ensemble_matching:  /media/cat/8TB/donato/bmi/DON-019683/20240124/ensembles/ensembles_matches_bmi_to_suite2p.npz\n",
      "idxs:  [ 5  9 20  0]\n",
      "fname_ensemble_matching:  /media/cat/8TB/donato/bmi/DON-019683/20240125/ensembles/ensembles_matches_bmi_to_suite2p.npz\n",
      "idxs:  [23 18 44 33]\n",
      "fname_ensemble_matching:  /media/cat/8TB/donato/bmi/DON-019683/20240126/ensembles/ensembles_matches_bmi_to_suite2p.npz\n",
      "idxs:  [12 14 12 10]\n",
      "fname_ensemble_matching:  /media/cat/8TB/donato/bmi/DON-019683/20240127/ensembles/ensembles_matches_bmi_to_suite2p.npz\n",
      "idxs:  [34 22  2 19]\n"
     ]
    }
   ],
   "source": [
    "#############################################################\n",
    "############# PAIRWISE CORRELATION DISTRIBUTIONS ############\n",
    "#############################################################\n",
    "from utils_calcium import run_pearson_corr_single_cell_vs_all_cells_parallel, compute_ensemble_only_correlations\n",
    "import parmap\n",
    "\n",
    "#\n",
    "compute_ensemble_only_correlations(c)\n",
    "\n",
    "#\n",
    "plt.figure(figsize=(10,10))\n",
    "for session_id in range(1,len(c.session_ids)):\n",
    "    fname_ensemble_matching = os.path.join(c.root_dir, \n",
    "                                        c.animal_id, \n",
    "                                        str(c.session_ids[session_id]), \n",
    "                                        'ensembles',\n",
    "                                        'ensembles_matches_bmi_to_suite2p.npz')\n",
    "    \n",
    "    print (\"fname_ensemble_matching: \", fname_ensemble_matching)\n",
    "    data = np.load(fname_ensemble_matching, allow_pickle=True)\n",
    "    #\n",
    "    idxs = []\n",
    "    idxs.append(data['idx_ensemble1_0'])\n",
    "    idxs.append(data['idx_ensemble1_1'])\n",
    "    idxs.append(data['idx_ensemble2_0'])\n",
    "    idxs.append(data['idx_ensemble2_1'])\n",
    "    idxs = np.array(idxs)\n",
    "    print (\"idxs: \", idxs)\n",
    "\n",
    "    # load the cell correlation vals \n",
    "    dir_corrs = os.path.join(c.root_dir,\n",
    "                                c.animal_id,\n",
    "                                str(c.session_ids[session_id]),\n",
    "                                'plane0',\n",
    "                                'correlations')\n",
    "    \n",
    "    # chekc if dir exists\n",
    "    if os.path.exists(dir_corrs)==False:\n",
    "        os.mkdir(dir_corrs)\n",
    "\n",
    "    # load the cell correlation vals using idxs as the ensemble\n",
    "    ctr=0\n",
    "    for idx in idxs:\n",
    "        fname_corr = os.path.join(dir_corrs, str(idx)+'.npz')\n",
    "\n",
    "        #\n",
    "        data = np.load(fname_corr, allow_pickle=True)\n",
    "        pc = data['pearson_corr']\n",
    "\n",
    "        # delete nans\n",
    "        idx = np.where(np.isnan(pc)==False)[0]\n",
    "        pc = pc[idx]\n",
    "\n",
    "        # make histogram of pc\n",
    "        y = np.histogram(pc, bins=np.arange(-0.2,0.5,0.02))\n",
    "        xx = y[1][:-1]\n",
    "        yy = y[0]\n",
    "\n",
    "        #\n",
    "        plt.title(str(roi_names[ctr]))\n",
    "\n",
    "        #\n",
    "        ax=plt.subplot(2,2,ctr+1)\n",
    "        #plt.semilogy()\n",
    "        plt.plot(xx, yy, linewidth=2, \n",
    "                 color=colors[session_id],\n",
    "                 label=str(c.session_ids[session_id]))\n",
    "\n",
    "        if ctr==0:\n",
    "            plt.legend() \n",
    "\n",
    "        #\n",
    "        plt.xlabel(\"Pearson correlation\")\n",
    "\n",
    "\n",
    "        #\n",
    "        ctr+=1\n",
    "\n",
    "#\n",
    "plt.suptitle(c.animal_id+\" \"+str(c.rec_type))\n",
    "\n",
    "plt.show()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "5b485d43",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2157941/787871892.py:6: MatplotlibDeprecationWarning: The get_cmap function was deprecated in Matplotlib 3.7 and will be removed two minor releases later. Use ``matplotlib.colormaps[name]`` or ``matplotlib.colormaps.get_cmap(obj)`` instead.\n",
      "  cmap = matplotlib.cm.get_cmap('viridis')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "idxs:  [ 8 15 11  1]\n",
      "nodes_list:  9\n",
      "important nodes:  [ 8 15 11  1]\n",
      "idxs:  [17 52 27  1]\n",
      "nodes_list:  29\n",
      "important nodes:  [17 52 27  1]\n",
      "idxs:  [10 23 12 55]\n",
      "nodes_list:  37\n",
      "important nodes:  [10 23 12 55]\n",
      "idxs:  [ 7 29 19  5]\n",
      "nodes_list:  17\n",
      "important nodes:  [ 7 29 19  5]\n",
      "idxs:  [ 5  9 20  0]\n",
      "nodes_list:  11\n",
      "important nodes:  [ 5  9 20  0]\n",
      "idxs:  [23 18 44 33]\n",
      "nodes_list:  10\n",
      "important nodes:  [23 18 44 33]\n",
      "idxs:  [12 14 12 10]\n",
      "nodes_list:  4\n",
      "important nodes:  [12 14 12 10]\n",
      "idxs:  [34 22  2 19]\n",
      "nodes_list:  6\n",
      "important nodes:  [34 22  2 19]\n"
     ]
    }
   ],
   "source": [
    "#############################################################\n",
    "############### GRAB NETWORK OF CELLS CONNECTED #############\n",
    "#############################################################\n",
    "\n",
    "# make viridis colormap of len(c.session_ids)\n",
    "cmap = matplotlib.cm.get_cmap('viridis')\n",
    "colors = cmap(np.linspace(0.1, 0.9, len(c.session_ids)))\n",
    "clrs = ['blue','lightblue','red','pink']\n",
    "roi_names = ['pos1','pos2','neg1','neg2']\n",
    "\n",
    "corr_thresh = 0.1\n",
    "corr_thresh_zscore = 5\n",
    "\n",
    "use_raw_pcorr = False\n",
    "\n",
    "#\n",
    "plt.figure(figsize=(10,10))\n",
    "for session_id in range(1,len(c.session_ids)):\n",
    "    fname_ensemble_matching = os.path.join(c.root_dir, \n",
    "                                        c.animal_id, \n",
    "                                        str(c.session_ids[session_id]), \n",
    "                                        'ensembles',\n",
    "                                        'ensembles_matches_bmi_to_suite2p.npz')\n",
    "    \n",
    "    #print (\"fname_ensemble_matching: \", fname_ensemble_matching)\n",
    "    data = np.load(fname_ensemble_matching, allow_pickle=True)\n",
    "    #\n",
    "    idxs = []\n",
    "    idxs.append(data['idx_ensemble1_0'])\n",
    "    idxs.append(data['idx_ensemble1_1'])\n",
    "    idxs.append(data['idx_ensemble2_0'])\n",
    "    idxs.append(data['idx_ensemble2_1'])\n",
    "    idxs = np.array(idxs)\n",
    "    print (\"idxs: \", idxs)\n",
    "\n",
    "    # load the cell correlation vals \n",
    "    dir_corrs = os.path.join(c.root_dir,\n",
    "                                c.animal_id,\n",
    "                                str(c.session_ids[session_id]),\n",
    "                                'plane0',\n",
    "                                'correlations')\n",
    "    \n",
    "    # load the cell correlation vals using idxs as the ensemble\n",
    "    fname_corr = os.path.join(dir_corrs, str(idxs[0])+'.npz')\n",
    "    data = np.load(fname_corr, allow_pickle=True)\n",
    "    pc = data['pearson_corr']\n",
    "    n_cells = pc.shape[0]\n",
    "    m = np.zeros((n_cells,n_cells), dtype=np.int32)\n",
    "\n",
    "    #\n",
    "    ctr=0\n",
    "    for idx in idxs:\n",
    "        fname_corr = os.path.join(dir_corrs, str(idx)+'.npz')\n",
    "        try:\n",
    "            data = np.load(fname_corr, allow_pickle=True)\n",
    "        except:\n",
    "            print (\"missing cell: \", idx)\n",
    "            continue\n",
    "        #\n",
    "        pc = data['pearson_corr']\n",
    "        pc_zscore = data['z_score_pearson_corr']\n",
    "\n",
    "        # find correlations > corr_thresh\n",
    "        if use_raw_pcorr:\n",
    "            idx_thresh = np.where(pc>corr_thresh)[0]\n",
    "        else:\n",
    "            idx_thresh = np.where(pc_zscore>corr_thresh_zscore)[0]\n",
    "\n",
    "        # \n",
    "        for id2 in idx_thresh:\n",
    "            m[idx, id2]=1\n",
    "            m[id2, idx]=1\n",
    "\n",
    "        ctr+=1          \n",
    "\n",
    "    #\n",
    "    ctr3 = 2\n",
    "    for idx in idxs:\n",
    "        m[idx,idx]= ctr3\n",
    "        ctr3+=1\n",
    "\n",
    "    #\n",
    "    G = nx.Graph(m)\n",
    "\n",
    "    # remove isolates\n",
    "    G.remove_nodes_from(list(nx.isolates(G)))\n",
    "\n",
    "    nodes_list = list(G.nodes)\n",
    "    print (\"nodes_list: \", len(nodes_list))\n",
    "    print (\"important nodes: \", idxs)\n",
    "\n",
    "    # make a color list all black except the important nodes\n",
    "    node_colors = []\n",
    "    ctr=0\n",
    "    for k in range(len(nodes_list)):\n",
    "        if nodes_list[k] in idxs:\n",
    "            node_colors.append(clrs[ctr])\n",
    "            ctr+=1\n",
    "        else:\n",
    "            node_colors.append('black')\n",
    "\n",
    "\n",
    "    ##############################################################\n",
    "    ax=plt.subplot(2,4,session_id)\n",
    "\n",
    "    # remove self loops\n",
    "    G.remove_edges_from(nx.selfloop_edges(G))\n",
    "\n",
    "    #\n",
    "    pos = nx.spring_layout(G)\n",
    "    nx.draw(G, \n",
    "            pos,\n",
    "            with_labels=False, \n",
    "            node_size=20,\n",
    "            node_color=node_colors\n",
    "            #font_weight='bold'\n",
    "            )\n",
    "\n",
    "    # add ylabel on the plot\n",
    "    ax.set_ylabel(str(session_id), labelpad=20)\n",
    "\n",
    "    #\n",
    "    ctr+=1\n",
    "    axis = plt.gca()\n",
    "    plt.box(True)\n",
    "\n",
    "    # draw a border around the figure\n",
    "    ax.set_title(str(c.session_types[session_id]), fontsize=20)\n",
    "\n",
    "#\n",
    "plt.suptitle(c.animal_id+\" \"+str(c.rec_type))\n",
    "\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "e2421a7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure()\n",
    "\n",
    "#\n",
    "ax = plt.subplot(411)\n",
    "idx = np.argmax(corr1[0])\n",
    "plt.plot(F_filtered[idx],c='black')\n",
    "temp = rois1[0]\n",
    "f0 = np.median(temp)\n",
    "temp = (temp-f0)/f0\n",
    "plt.plot(temp,c='blue',\n",
    "         alpha=.5)\n",
    "\n",
    "#\n",
    "ax = plt.subplot(412)\n",
    "idx = np.argmax(corr1[1])\n",
    "plt.plot(F_filtered[idx],c='black')\n",
    "temp = rois1[1]\n",
    "f0 = np.median(temp)\n",
    "temp = (temp-f0)/f0\n",
    "plt.plot(temp,c='blue',\n",
    "         alpha=.5)\n",
    "\n",
    "#\n",
    "ax = plt.subplot(413)\n",
    "idx = np.argmax(corr2[0])\n",
    "plt.plot(F_filtered[idx],c='black')\n",
    "temp = rois2[0]\n",
    "f0 = np.median(temp)\n",
    "temp = (temp-f0)/f0\n",
    "plt.plot(temp,c='blue'\n",
    "         ,alpha=.5)\n",
    "\n",
    "#\n",
    "ax = plt.subplot(414)\n",
    "idx = np.argmax(corr2[1])\n",
    "plt.plot(F_filtered[idx],c='black')\n",
    "temp = rois2[1]\n",
    "f0 = np.median(temp)\n",
    "temp = (temp-f0)/f0\n",
    "plt.plot(temp,c='blue'\n",
    "         ,alpha=.5)\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5777238",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a28f8e4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "9cf196a4",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/5 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 5/5 [01:40<00:00, 20.16s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DONE\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "import networkx as nx\n",
    "import pickle\n",
    "import yaml\n",
    "\n",
    "\n",
    "#\n",
    "root_dir = '/media/cat/8TB/donato/bmi/'\n",
    "\n",
    "#\n",
    "pcorr_thresh = 5\n",
    "recompute_graphs = True\n",
    "use_median = True\n",
    "\n",
    "#\n",
    "for animal_id in tqdm(animal_ids):\n",
    "\n",
    "    # load yaml file and get session_ids field\n",
    "    fname_yaml = os.path.join(root_dir, \n",
    "                              animal_id, \n",
    "                              str(animal_id)+'.yaml')\n",
    "    \n",
    "    # load yaml file\n",
    "    d = yaml.load(open(fname_yaml), \n",
    "                  Loader=yaml.FullLoader)\n",
    "    \n",
    "    # get session ids\n",
    "    session_ids = d['session_ids']\n",
    "\n",
    "\n",
    "\n",
    "    # load all the pairwise distributions\n",
    "    diameters = []\n",
    "    clustering = []\n",
    "    aspl = []\n",
    "    degree = []\n",
    "    pagerank = []\n",
    "    ctr = 0\n",
    "    for session_id in session_ids:\n",
    "        \n",
    "        #\n",
    "        dir_main = os.path.join(root_dir, \n",
    "                                animal_id, \n",
    "                                str(session_id),\n",
    "                                'plane0',\n",
    "                                #'merged',\n",
    "                                'correlations')\n",
    "        \n",
    "        #\n",
    "        fname_out = os.path.join(dir_main, 'graph_metrics.npz')\n",
    "       #print (\"fname_out: \", fname_out)\n",
    "\n",
    "        if os.path.exists(fname_out)==False or recompute_graphs:\n",
    "\n",
    "            # load all the .npz files from dir_main\n",
    "            files = os.listdir(dir_main)\n",
    "            # delte all files that have \"graph\" in filename\n",
    "            files = [file for file in files if 'graph' not in file]\n",
    "            #print (\"# files: \", len(files))\n",
    "\n",
    "            # initialize connectivity matrix\n",
    "            cm = np.zeros((len(files), len(files)), 'float32')\n",
    "\n",
    "            #\n",
    "            for k in range(len(files)):\n",
    "\n",
    "                # load cell correlations\n",
    "                fname_pcorr = os.path.join(dir_main, files[k])\n",
    "                d = np.load(fname_pcorr, allow_pickle=True)\n",
    "                id = d['id']\n",
    "                zscore = d['z_score_pearson_corr']\n",
    "\n",
    "                # find which values are above threshold\n",
    "                idx = np.where(zscore>pcorr_thresh)[0]\n",
    "                cm[k,idx] = zscore[idx]\n",
    "\n",
    "                # this is a symmetric matrix\n",
    "                cm[idx,k] = zscore[idx]\n",
    "\n",
    "\n",
    "            # make a network graph from the cm matrix\n",
    "            G = nx.Graph(cm)\n",
    "\n",
    "            # remove self connections\n",
    "            G.remove_edges_from(nx.selfloop_edges(G))\n",
    "\n",
    "            # remove isolates\n",
    "            G.remove_nodes_from(list(nx.isolates(G)))\n",
    "\n",
    "            # find largest component in G\n",
    "            G_largest = G.subgraph(max(nx.connected_components(G), key=len))\n",
    "\n",
    "            # compute the diameter of G\n",
    "            diam = nx.diameter(G_largest)\n",
    "\n",
    "            # grab the clustering coefficient of G for all nodes\n",
    "            clust = nx.clustering(G_largest)\n",
    "            # compute the mean clustering coefficient from clust dict\n",
    "            if use_median:\n",
    "                clust = np.median(list(clust.values()))\n",
    "            else:\n",
    "                clust = np.mean(list(clust.values()))\n",
    "\n",
    "            # compute the average shortest path length\n",
    "            path_len = nx.average_shortest_path_length(G_largest)\n",
    "                \n",
    "            # compute the average degree\n",
    "            deg = nx.average_degree_connectivity(G_largest)\n",
    "            if use_median:\n",
    "                deg = np.median(list(deg.values()))\n",
    "            else:\n",
    "                deg = np.mean(list(deg.values()))\n",
    "\n",
    "            # compute the pagerank\n",
    "            pr = nx.pagerank(G_largest)\n",
    "            if use_median:\n",
    "                pr = np.median(list(pr.values()))\n",
    "            else:\n",
    "                pr = np.mean(list(pr.values()))\n",
    "\n",
    "            # Save the graph to a Pickle file\n",
    "            fname_pkl = os.path.join(dir_main, 'graph.pkl')\n",
    "            with open(fname_pkl, \"wb\") as f:\n",
    "                pickle.dump(G, f)\n",
    "        \n",
    "            np.savez(fname_out,\n",
    "                    diameters = diam,\n",
    "                    clustering = clust,\n",
    "                    aspl = path_len,\n",
    "                    degree = deg,\n",
    "                    pagerank = pr,\n",
    "                    pcorr_thresh = pcorr_thresh)\n",
    "\n",
    "\n",
    "        else:\n",
    "            d = np.load(fname_out, allow_pickle=True)\n",
    "            diam = d['diameters']\n",
    "            clust = d['clustering']\n",
    "            path_len = d['aspl']\n",
    "            deg = d['degree']\n",
    "            pr = d['pagerank']\n",
    "\n",
    "            #\n",
    "            G = pickle.load(open(os.path.join(os.path.split(fname_out)[0], \n",
    "                                              'graph.pkl')), \"rb\")\n",
    "\n",
    "        # add to lists\n",
    "        diameters.append(diam)\n",
    "        clustering.append(clust)\n",
    "        aspl.append(path_len)\n",
    "        degree.append(deg)\n",
    "        pagerank.append(pr)\n",
    "    \n",
    "        #\n",
    "        ctr+=1\n",
    "\n",
    "    # save master list\n",
    "    dir_out = os.path.join(root_dir, \n",
    "                           animal_id, \n",
    "                           'correlations')\n",
    "    if os.path.exists(dir_out)==False:\n",
    "        os.mkdir(dir_out)\n",
    "\n",
    "    # save master list\n",
    "    # load yaml file and get session_ids field\n",
    "    fname_out_animal = os.path.join(root_dir, \n",
    "                                    animal_id, \n",
    "                                    'correlations',\n",
    "                                    'graph_metrics.npz')\n",
    "    \n",
    "    np.savez(fname_out_animal,\n",
    "            pcorr_thresh = pcorr_thresh,\n",
    "            diameters = diameters,\n",
    "            clustering = clustering,\n",
    "            aspl = aspl,\n",
    "            degree = degree,\n",
    "            pagerank = pagerank\n",
    "            )\n",
    "\n",
    "print (\"DONE\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "a91304ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/cat/.conda/envs/bmitools/lib/python3.8/site-packages/scipy/stats/_stats_py.py:4424: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  warnings.warn(stats.ConstantInputWarning(msg))\n"
     ]
    }
   ],
   "source": [
    "#\n",
    "def proc_data(ax, \n",
    "              data,\n",
    "              data_name,\n",
    "              clr,\n",
    "              text=None):\n",
    "\n",
    "    ax.set_title(data_name)\n",
    "    ax.plot(data,\n",
    "             linewidth=2,\n",
    "             color=clr)\n",
    "    # fit linear regression to the data\n",
    "    x = np.arange(len(data))\n",
    "    y = np.array(data)\n",
    "    idx = np.isfinite(y)\n",
    "    z = np.polyfit(x[idx], y[idx], 1)\n",
    "    p = np.poly1d(z)\n",
    "\n",
    "    rr,pp = scipy.stats.pearsonr(x,y)\n",
    "    #\n",
    "    ax.plot(x,p(x),\n",
    "             '--',\n",
    "             linewidth=2,\n",
    "             c=clr,\n",
    "             label = ('r='+str(np.round(rr,2))+' p='+str(np.round(pp,2))) if text is None else text\n",
    "             )\n",
    "    \n",
    "\n",
    "#\n",
    "clrs = ['blue','navy','lightblue','red','orange']\n",
    "\n",
    "# plot the distributions\n",
    "plt.figure(figsize=(10,10))\n",
    "plt.suptitle(c.animal_id)\n",
    "\n",
    "\n",
    "\n",
    "#\n",
    "for ctr, animal_id in enumerate(animal_ids):\n",
    "    fname = os.path.join(c.root_dir, \n",
    "                        animal_id,\n",
    "                        'correlations',\n",
    "                        'graph_metrics.npz')\n",
    "\n",
    "    #\n",
    "    data = np.load(fname, allow_pickle=True)\n",
    "    diameters = data['diameters']\n",
    "    clustering = data['clustering']\n",
    "    aspl = data['aspl']\n",
    "    degree = data['degree']\n",
    "    pagerank = data['pagerank']\n",
    "\n",
    "    # \n",
    "    data_array = [diameters, clustering, aspl, degree, pagerank]\n",
    "    data_names = ['Diameter', 'Clustering', 'Ave path len', 'Degree', 'Pagerank']\n",
    "\n",
    "    #\n",
    "    for k in range(5):\n",
    "        ax = plt.subplot(2,3,k+1)\n",
    "\n",
    "        proc_data(ax,\n",
    "                    data_array[k],\n",
    "                    data_names[k],\n",
    "                    clrs[ctr],\n",
    "                    text = animal_id if k==0 else None\n",
    "                    )\n",
    "        ax.legend()\n",
    "\n",
    "        # label xaxis with session starting with 'day0' and followed by 'bmiN' where n\n",
    "        ax.set_xlabel('Session #')\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "id": "61ebc2a3",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1215065/82470050.py:26: MatplotlibDeprecationWarning: The get_cmap function was deprecated in Matplotlib 3.7 and will be removed two minor releases later. Use ``matplotlib.colormaps[name]`` or ``matplotlib.colormaps.get_cmap(obj)`` instead.\n",
      "  cmap = matplotlib.cm.get_cmap('viridis')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# cells:  540\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n",
      "Error []\n"
     ]
    }
   ],
   "source": [
    "# plot the single cell correlations over time\n",
    "#\n",
    "animal_id = 'DON-015801'\n",
    "animal_id = 'DON-011733'\n",
    "animal_ids = [\n",
    "    \"DON-011733\",    # M1\n",
    "    \"DON-014618\",    # M1\n",
    "    'DON-014451',    # M1\n",
    "    'DON-014371',    # CA3\n",
    "    'DON-015801',    # CA3\n",
    "]\n",
    "\n",
    "animal_id = animal_ids[1]\n",
    "\n",
    "# load the yaml file to get session_ids\n",
    "fname_yaml = os.path.join(root_dir, \n",
    "                          animal_id, \n",
    "                          str(animal_id)+'.yaml')\n",
    "\n",
    "# load yaml file\n",
    "d = yaml.load(open(fname_yaml), \n",
    "              Loader=yaml.FullLoader)\n",
    "session_ids = d['session_ids']\n",
    "\n",
    "# make viridis map over len(session_ids)\n",
    "cmap = matplotlib.cm.get_cmap('viridis')\n",
    "colors = [cmap(i) for i in np.linspace(0, 1, len(session_ids))]\n",
    "\n",
    "# find # of cells in recording\n",
    "fname = os.path.join(root_dir,\n",
    "                    animal_id,\n",
    "                    str(session_ids[0]),\n",
    "                    'plane0',\n",
    "                    'merged',\n",
    "                    'F.npy')\n",
    "n_cells = np.load(fname).shape[0]\n",
    "print (\"# cells: \", n_cells)\n",
    "\n",
    "\n",
    "#\n",
    "cell_modes = []\n",
    "for cell_id in range(n_cells):\n",
    "\n",
    "    #\n",
    "    temp_mode = []\n",
    "    for ctr, session_id in enumerate(session_ids):\n",
    "        fname = os.path.join(root_dir,\n",
    "                            animal_id,\n",
    "                            str(session_id),\n",
    "                            'plane0',\n",
    "                           # 'merged',\n",
    "                            'correlations',\n",
    "                            str(cell_id)+'.npz')\n",
    "        \n",
    "        #\n",
    "        try:\n",
    "            dd = np.load(fname, allow_pickle=True)\n",
    "        except:\n",
    "            continue\n",
    "\n",
    "        pcorr = dd['pearson_corr']\n",
    "\n",
    "        # remove nans\n",
    "        idx = np.isnan(pcorr)==False\n",
    "        pcorr = pcorr[idx]\n",
    "        # # get mode of pcor\n",
    "\n",
    "        # find 90th percentile of the pcorr distrib\n",
    "        try:\n",
    "            percentile = np.percentile(pcorr, 90)\n",
    "        except:\n",
    "            print (\"Error\", pcorr)\n",
    "\n",
    "        temp_mode.append(percentile)  \n",
    "\n",
    "        #break\n",
    "    #\n",
    "    cell_modes.append(temp_mode)\n",
    "\n",
    "    #break\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "id": "d90407a9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "540\n",
      "9\n"
     ]
    }
   ],
   "source": [
    "#\n",
    "print (len(cell_modes))\n",
    "print (len(cell_modes[0]))\n",
    "\n",
    "\n",
    "# plot violion plots using seaborn\n",
    "import seaborn as sns\n",
    "import pandas as pd\n",
    "\n",
    "#\n",
    "n_sess = 9\n",
    "sns.set(style=\"whitegrid\")\n",
    "plt.figure(figsize=(8, 4))\n",
    "sns.violinplot(data=cell_modes[:n_sess], \n",
    "               #inner=\"quart\", \n",
    "               #palette=\"Set3\"\n",
    "               )\n",
    "\n",
    "#\n",
    "sess = ['day0', 'bmi1', 'bmi2', 'bmi3', 'bmi4', 'bmi5', 'bmi6', 'bmi7', 'bmi8']\n",
    "plt.xticks(np.arange(n_sess), sess[:n_sess], rotation=0)\n",
    "\n",
    "plt.ylabel('Median pairwise correlation')\n",
    "\n",
    "plt.suptitle(animal_id)\n",
    "\n",
    "#\n",
    "plt.show()\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "33656bbe",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "1c9f393c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1215065/936415909.py:16: MatplotlibDeprecationWarning: The get_cmap function was deprecated in Matplotlib 3.7 and will be removed two minor releases later. Use ``matplotlib.colormaps[name]`` or ``matplotlib.colormaps.get_cmap(obj)`` instead.\n",
      "  cmap = matplotlib.cm.get_cmap('viridis')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# cells:  712\n"
     ]
    }
   ],
   "source": [
    "# plot the single cell correlations over time\n",
    "\n",
    "animal_id = 'DON-015801'\n",
    "\n",
    "# load the yaml file to get session_ids\n",
    "fname_yaml = os.path.join(root_dir, \n",
    "                          animal_id, \n",
    "                          str(animal_id)+'.yaml')\n",
    "\n",
    "# load yaml file\n",
    "d = yaml.load(open(fname_yaml), \n",
    "              Loader=yaml.FullLoader)\n",
    "session_ids = d['session_ids']\n",
    "\n",
    "# make viridis map over len(session_ids)\n",
    "cmap = matplotlib.cm.get_cmap('viridis')\n",
    "colors = [cmap(i) for i in np.linspace(0, 1, len(session_ids))]\n",
    "\n",
    "# find # of cells in recording\n",
    "fname = os.path.join(root_dir,\n",
    "                    animal_id,\n",
    "                    str(session_ids[0]),\n",
    "                    'plane0',\n",
    "                    'merged',\n",
    "                    'F.npy')\n",
    "n_cells = np.load(fname).shape[0]\n",
    "print (\"# cells: \", n_cells)\n",
    "\n",
    "\n",
    "#\n",
    "n_plots = 100\n",
    "plt.figure(figsize=(15,15))\n",
    "ctra =0\n",
    "#cell_ids = np.arange()\n",
    "cell_modes = []\n",
    "for cell_id in range(n_cells):\n",
    "    ax = plt.subplot(int(np.sqrt(n_plots)),\n",
    "                     int(np.sqrt(n_plots)),\n",
    "                     ctra+1)\n",
    "\n",
    "    #\n",
    "    temp_mode = []\n",
    "    for ctr, session_id in enumerate(session_ids):\n",
    "        fname = os.path.join(root_dir,\n",
    "                            animal_id,\n",
    "                            str(session_id),\n",
    "                            'plane0',\n",
    "                            'merged',\n",
    "                            'correlations',\n",
    "                            str(cell_id)+'.npz')\n",
    "        \n",
    "        #\n",
    "        try:\n",
    "            dd = np.load(fname, allow_pickle=True)\n",
    "        except:\n",
    "            continue\n",
    "\n",
    "        pcorr = dd['pearson_corr']\n",
    "        #print (pcorr)\n",
    "\n",
    "        # make a histogram\n",
    "        # y = np.histogram(pcorr, bins=np.arange(-0.1,0.25,0.01))   M1 mice\n",
    "        y = np.histogram(pcorr, bins=np.arange(-0.03,0.2,0.003))\n",
    "        xx = y[1][:-1]\n",
    "        yy = y[0]\n",
    "\n",
    "        # plot the histogram\n",
    "        plt.plot(xx,yy, \n",
    "                color=colors[ctr],\n",
    "                alpha=0.75\n",
    "                #linewidth=2\n",
    "                )\n",
    "        plt.yticks([])\n",
    "\n",
    "        plt.title(str(cell_id), fontsize=6)\n",
    "\n",
    "        # \n",
    "        if ctra<90:\n",
    "            plt.xticks([])\n",
    "        else:\n",
    "            # make tick size smaller\n",
    "            plt.xticks(fontsize=6)\n",
    "    #\n",
    "    ctra+=1\n",
    "\n",
    "    if ctra==n_plots:\n",
    "        plt.xlabel('Pearson correlation')\n",
    "        plt.ylabel('# pairs')\n",
    "        fname = os.path.join(root_dir,\n",
    "                            animal_id,\n",
    "                            'correlations',\n",
    "                            str(cell_id-n_plots+1)+'_correlations.png')\n",
    "        plt.savefig(fname,dpi=300)\n",
    "        plt.close()\n",
    "\n",
    "        #\n",
    "        plt.figure(figsize=(15,15))\n",
    "        ctra=0\n",
    "\n",
    "\n",
    "#\n",
    "plt.xlabel('Pearson correlation')\n",
    "plt.ylabel('# pairs')\n",
    "fname = os.path.join(root_dir,\n",
    "                    animal_id,\n",
    "                    'correlations',\n",
    "                    str(cell_id-n_plots+1)+'_correlations.png')\n",
    "plt.savefig(fname,dpi=300)\n",
    "plt.close()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48783c8e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4578927",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c099b02",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "3979279d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# files:  472\n"
     ]
    }
   ],
   "source": [
    "#\n",
    "# Load data from specific animal / day\n",
    "animal_id = 'DON-015801'\n",
    "session_id = '20230615'\n",
    "\n",
    "dir_main = os.path.join(root_dir,\n",
    "                        animal_id,\n",
    "                        session_id,\n",
    "                        'plane0',\n",
    "                        'correlations')\n",
    "\n",
    "# find all.npz files in dir_main\n",
    "files = os.listdir(dir_main)\n",
    "files = [file for file in files if 'graph' not in file]\n",
    "print (\"# files: \", len(files))\n",
    "\n",
    "# make a connectivity matrix\n",
    "cm = np.zeros((len(files), len(files)), 'float32')\n",
    "\n",
    "#\n",
    "pcorr_thresh = 5\n",
    "for k in range(len(files)):\n",
    "        \n",
    "    # load cell correlations\n",
    "    fname_pcorr = os.path.join(dir_main, files[k])\n",
    "    d = np.load(fname_pcorr, allow_pickle=True)\n",
    "    id = d['id']\n",
    "    zscore = d['z_score_pearson_corr']\n",
    "    \n",
    "    # find which values are above threshold\n",
    "    idx = np.where(zscore>pcorr_thresh)[0]\n",
    "    cm[k,idx] = zscore[idx]\n",
    "    \n",
    "    # this is a symmetric matrix\n",
    "    cm[idx,k] = zscore[idx]\n",
    "\n",
    "    # make a boolean matrix\n",
    "    cm_bool = np.zeros((cm.shape),'bool')\n",
    "    cm_bool[cm>0]=1\n",
    "\n",
    "\n",
    "#\n",
    "plt.figure()\n",
    "plt.imshow(cm_bool,\n",
    "           vmin=0,\n",
    "           vmax=1,\n",
    "           cmap='binary')\n",
    "plt.ylim(0,cm_bool.shape[0])\n",
    "plt.colorbar()\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ce4d147",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a14d395c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9c531e0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0329a0af",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "842e8835",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "18a1138c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2af732e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "000f3c04",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "0d1ca6ae",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "processing session:  20230130\n",
      " ... adding BLANK tone_state\n",
      " ... adding BLANK ensemble_state\n",
      " ... adding COPY OF post_reward_state as reward_state\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 60%|██████    | 54123/89951 [00:00<00:00, 270877.34it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... found rewarded trial at frame:  901  time:  30.033333333333335 s\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 89951/89951 [00:00<00:00, 269949.99it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# of trials:  1 # of rewarded trials:  1\n",
      "reward_times and trials[idx][:,1] are not identical\n",
      "reward_times:  [ 8787 33064]\n",
      "trials[idx][:,1]:  [901]\n",
      "Exiting...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "########################################################\n",
    "########### LOAD SPREADSHEETS AND METADATA #############\n",
    "########################################################\n",
    "\n",
    "# adds trial information; eventually should also add bursts etc...\n",
    "c.fix_spreadsheet()\n",
    "\n",
    "# \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e0ef26b2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0 0 0 ... 0 0 0]\n"
     ]
    }
   ],
   "source": [
    "print (c.reward_state )\n",
    "\n",
    "plt.figure()\n",
    "plt.plot(c.reward_state)\n",
    "plt.plot(c.post_reward_state)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "05ac2aa2",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# \n",
    "print (c.session_ids)\n",
    "c.fix_spreadsheet_missing_vals = True\n",
    "\n",
    "#\n",
    "print (c.session_types)\n",
    "c.load_spreadsheet_multi()\n",
    "\n",
    "# also save the res array in the same folder\n",
    "c.load_best_matches()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "b9e0df1c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# bouts to be zeroed out:  3\n"
     ]
    }
   ],
   "source": [
    "########################################################\n",
    "########## LOAD PAIRWISE CORRELATION DATA ##############\n",
    "########################################################\n",
    "# this is precomputed on the cluster \n",
    "#\n",
    "def zero_ca_artifacts(fname):\n",
    "    d = np.load(fname, \n",
    "                allow_pickle=True)\n",
    "\n",
    "    # load the upphase_bin\n",
    "    F_upphase = d['F_upphase']\n",
    "    F_detrended = d['F_detrended']\n",
    "    F_filtered = d['F_filtered']\n",
    " \n",
    "\n",
    "    # make mua\n",
    "    mua = np.sum(F_upphase,0)\n",
    "\n",
    "    thresh = 60\n",
    "    time_thresh = 30*2\n",
    "\n",
    "    # find periods where mua > thresh and that are longer than 10 frames\n",
    "    idx = np.where(mua>thresh)[0]\n",
    "\n",
    "    # find contiguous periods within idx and measure duration\n",
    "    bouts = []\n",
    "    start = idx[0]\n",
    "    #\n",
    "    for k in range(1,len(idx)-1,1):\n",
    "        if idx[k]==idx[k+1]-1:\n",
    "            continue\n",
    "        else:\n",
    "            end = idx[k]\n",
    "            if (end-start)>time_thresh:\n",
    "                bouts.append([start, end])\n",
    "            start = idx[k+1]\n",
    "\n",
    "    #\n",
    "    bouts = np.array(bouts)\n",
    "    print (\"# bouts to be zeroed out: \", bouts.shape[0])    \n",
    "\n",
    "    #\n",
    "    for bout in bouts:\n",
    "        F_upphase[:,bout[0]:bout[1]]=0\n",
    "        F_detrended[:,bout[0]:bout[1]]=0\n",
    "        F_filtered[:,bout[0]:bout[1]]=0\n",
    "\n",
    "    # save thsi files separately for each session\n",
    "    fname_out = os.path.join(os.path.split(fname)[0],\n",
    "                             'F_upphase.npy')\n",
    "    np.save(fname_out, F_upphase)\n",
    "\n",
    "    fname_out = os.path.join(os.path.split(fname)[0],\n",
    "                                'F_detrended.npy')\n",
    "    np.save(fname_out, F_detrended)\n",
    "\n",
    "    fname_out = os.path.join(os.path.split(fname)[0],\n",
    "                                'F_filtered.npy')\n",
    "    np.save(fname_out, F_filtered)\n",
    "\n",
    "    \n",
    "\n",
    "#\n",
    "fname = '/media/cat/8TB/donato/bmi/DON-015801/20230614/plane0/binarized_traces.npz'\n",
    "fname = '/media/cat/8TB/donato/bmi/DON-015801/20230614/plane0/merged/binarized_traces.npz'\n",
    "zero_ca_artifacts(fname)\n",
    "\n",
    "#\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1819f479",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c94f5dc2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0a483137",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5a9b03df",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "136731fd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41556ae3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "419d62bc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "c.cells_to_plot:  [270 321 223 196]\n",
      "session:  1  of 8 ,  reward times:  (33,)\n",
      "session:  2  of 8 ,  reward times:  (41,)\n",
      "session:  3  of 8 ,  reward times:  (33,)\n",
      "session:  4  of 8 ,  reward times:  (45,)\n",
      "session:  5  of 8 ,  reward times:  (64,)\n",
      "session:  6  of 8 ,  reward times:  (39,)\n",
      "session:  7  of 8 ,  reward times:  (56,)\n",
      "session:  8  of 8 ,  reward times:  (56,)\n"
     ]
    }
   ],
   "source": [
    "##############################################################\n",
    "######## COMPUTE REWARD TRIGGERED RESPONSE OF ROIs ###########\n",
    "##############################################################\n",
    "# \n",
    "from utils_calcium import plot_psth_rois_only\n",
    "\n",
    "#\n",
    "def init_fig():\n",
    "    plt.figure(figsize=(20,12))\n",
    "    axes = []\n",
    "    for k in range(4):\n",
    "        ax=plt.subplot(2,2,k+1)\n",
    "        axes.append(ax) \n",
    "\n",
    "    return axes\n",
    "\n",
    "# \n",
    "window = 300\n",
    "start_cell=0\n",
    "show_random = True\n",
    "idx_cells = None\n",
    "smoothing = True\n",
    "plot_separately = False\n",
    "c.plot_rois_only = True\n",
    "c.cells_to_plot = c.best_matches\n",
    "c.cells_to_plot = np.random.choice(np.arange(c.sessions[0].F_filtered.shape[0]),\n",
    "                                   4,\n",
    "                                   replace=False)\n",
    "#\n",
    "print (\"c.cells_to_plot: \", c.cells_to_plot)\n",
    "#\n",
    "axes = init_fig()\n",
    "\n",
    "#\n",
    "for session_id in np.arange(1,9,1):\n",
    "\n",
    "    #\n",
    "    psths_avg, psths_shuffled_avg, idx_cells, n_bursts  = get_reward_triggered_psth(session_id, \n",
    "                                                                                    c,\n",
    "                                                                                    window,\n",
    "                                                                                    idx_cells)\n",
    "    #\n",
    "    plot_psth_rois_only(c,\n",
    "                        psths_avg,\n",
    "                        psths_shuffled_avg,\n",
    "                        session_id,\n",
    "                        axes,\n",
    "                        start_cell,\n",
    "                        idx_cells,\n",
    "                        show_random,\n",
    "                        smoothing,\n",
    "                        window\n",
    "                        )\n",
    "\n",
    "\n",
    "    # reset axes\n",
    "    if plot_separately:\n",
    "        #\n",
    "        plt.suptitle(c.animal_id+\" session: \"+str(session_id),fontsize=20)\n",
    "        #\n",
    "        plt.savefig('/home/cat/'+str(start_cell)+\"_session_\"+str(session_id)+'.png',dpi=300)\n",
    "        plt.close()\n",
    "\n",
    "        axes = init_fig()\n",
    "\n",
    "#\n",
    "plt.suptitle(c.animal_id+\" session: \"+str(session_id),fontsize=20)\n",
    "#\n",
    "plt.savefig('/home/cat/'+str(start_cell)+\"_session_\"+str(session_id)+'.png',dpi=300)\n",
    "plt.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "6f8ab619",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "session:  1  of 8 ,  reward times:  (33,)\n",
      "session:  2  of 8 ,  reward times:  (41,)\n",
      "session:  3  of 8 ,  reward times:  (33,)\n",
      "session:  4  of 8 ,  reward times:  (45,)\n",
      "session:  5  of 8 ,  reward times:  (64,)\n",
      "session:  6  of 8 ,  reward times:  (39,)\n",
      "session:  7  of 8 ,  reward times:  (56,)\n",
      "session:  8  of 8 ,  reward times:  (56,)\n",
      "(8, 540, 360)\n"
     ]
    }
   ],
   "source": [
    "##############################################################\n",
    "######## COMPUTE REWARD TRIGGERED RESPONSE OF CELLS ##########\n",
    "##############################################################\n",
    "window = 30*6\n",
    "psth_array = []\n",
    "idx_cells = None\n",
    "global_order = False\n",
    "session_ids = np.arange(1,len(c.session_ids),1)\n",
    "\n",
    "#\n",
    "n_bursts_array = []\n",
    "for session_id in session_ids:\n",
    "#for session_id in [1,7]:\n",
    "\n",
    "    #\n",
    "    psths_avg, psths_shuffled_avg, idx_cells, n_bursts  = get_reward_triggered_psth(session_id, \n",
    "                                                                                    c,\n",
    "                                                                                    window,\n",
    "                                                                                    idx_cells)\n",
    "    \n",
    "    #\n",
    "    n_bursts_array.append(n_bursts)\n",
    "    \n",
    "    #\n",
    "    psth_array.append(psths_avg[idx_cells[::-1]])\n",
    "\n",
    "    if global_order==False:\n",
    "        idx_cells = None\n",
    "\n",
    "#\n",
    "psth_array = np.array(psth_array)\n",
    "print (psth_array.shape)\n",
    "\n",
    "#\n",
    "vmax = 0.3\n",
    "sua = plot_multi_session_psth_imshow(c,\n",
    "                                session_ids,\n",
    "                                psth_array,\n",
    "                                window,\n",
    "                                n_bursts_array,\n",
    "                                vmax=vmax)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4f6388b5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "572f0a29",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "low pass filter: 100%|██████████| 1/1 [00:00<00:00, 531.80it/s]\n",
      "model filter: remove bleaching or trends: 100%|██████████| 1/1 [00:00<00:00, 528.05it/s]\n",
      "fitting mode to physics:   0%|          | 0/1 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/cat/.conda/envs/bmi/lib/python3.8/site-packages/binarize2pcalcium/binarize2pcalcium.py:4026: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.\n",
      "  y_mode = scipy.stats.mode(F_filtered2)[0]\n",
      "fitting mode to physics: 100%|██████████| 1/1 [00:00<00:00, 81.99it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cumsum:  [4.63649033e-15 9.27561305e-15 1.39173696e-14 ... 1.00000000e+00\n",
      " 1.00000000e+00 1.00000000e+00]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "binarizing continuous traces filtered fluorescence onphase: 100%|██████████| 1/1 [00:00<00:00, 79.64it/s]\n",
      "binarizing continuous traces filtered fluorescence upphase: 100%|██████████| 1/1 [00:00<00:00, 74.52it/s]\n",
      "low pass filter: 100%|██████████| 1/1 [00:00<00:00, 560.59it/s]\n",
      "model filter: remove bleaching or trends: 100%|██████████| 1/1 [00:00<00:00, 561.79it/s]\n",
      "fitting mode to physics: 100%|██████████| 1/1 [00:00<00:00, 76.03it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cumsum:  [5.01316877e-10 1.00276836e-09 1.50435447e-09 ... 1.00000000e+00\n",
      " 1.00000000e+00 1.00000000e+00]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "binarizing continuous traces filtered fluorescence onphase: 100%|██████████| 1/1 [00:00<00:00, 61.34it/s]\n",
      "binarizing continuous traces filtered fluorescence upphase: 100%|██████████| 1/1 [00:00<00:00, 51.25it/s]\n",
      "low pass filter: 100%|██████████| 1/1 [00:00<00:00, 592.58it/s]\n",
      "model filter: remove bleaching or trends: 100%|██████████| 1/1 [00:00<00:00, 561.79it/s]\n",
      "fitting mode to physics: 100%|██████████| 1/1 [00:00<00:00, 80.49it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cumsum:  [2.27001954e-13 4.54110149e-13 6.81324632e-13 ... 1.00000000e+00\n",
      " 1.00000000e+00 1.00000000e+00]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "binarizing continuous traces filtered fluorescence onphase: 100%|██████████| 1/1 [00:00<00:00, 89.88it/s]\n",
      "binarizing continuous traces filtered fluorescence upphase: 100%|██████████| 1/1 [00:00<00:00, 76.70it/s]\n",
      "low pass filter: 100%|██████████| 1/1 [00:00<00:00, 607.69it/s]\n",
      "model filter: remove bleaching or trends: 100%|██████████| 1/1 [00:00<00:00, 556.42it/s]\n",
      "fitting mode to physics: 100%|██████████| 1/1 [00:00<00:00, 72.33it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cumsum:  [5.99037350e-12 1.19830454e-11 1.79780164e-11 ... 1.00000000e+00\n",
      " 1.00000000e+00 1.00000000e+00]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "binarizing continuous traces filtered fluorescence onphase: 100%|██████████| 1/1 [00:00<00:00, 60.19it/s]\n",
      "binarizing continuous traces filtered fluorescence upphase: 100%|██████████| 1/1 [00:00<00:00, 50.29it/s]\n",
      "low pass filter: 100%|██████████| 1/1 [00:00<00:00, 585.31it/s]\n",
      "model filter: remove bleaching or trends: 100%|██████████| 1/1 [00:00<00:00, 555.32it/s]\n",
      "fitting mode to physics: 100%|██████████| 1/1 [00:00<00:00, 68.69it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cumsum:  [3.64828491e-11 7.29779879e-11 1.09485420e-10 ... 1.00000000e+00\n",
      " 1.00000000e+00 1.00000000e+00]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "binarizing continuous traces filtered fluorescence onphase: 100%|██████████| 1/1 [00:00<00:00, 61.93it/s]\n",
      "binarizing continuous traces filtered fluorescence upphase: 100%|██████████| 1/1 [00:00<00:00, 53.58it/s]\n",
      "low pass filter: 100%|██████████| 1/1 [00:00<00:00, 623.60it/s]\n",
      "model filter: remove bleaching or trends: 100%|██████████| 1/1 [00:00<00:00, 600.30it/s]\n",
      "fitting mode to physics: 100%|██████████| 1/1 [00:00<00:00, 76.58it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cumsum:  [1.26496650e-17 2.53084175e-17 3.79762638e-17 ... 1.00000000e+00\n",
      " 1.00000000e+00 1.00000000e+00]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "binarizing continuous traces filtered fluorescence onphase: 100%|██████████| 1/1 [00:00<00:00, 63.18it/s]\n",
      "binarizing continuous traces filtered fluorescence upphase: 100%|██████████| 1/1 [00:00<00:00, 56.17it/s]\n",
      "low pass filter: 100%|██████████| 1/1 [00:00<00:00, 595.87it/s]\n",
      "model filter: remove bleaching or trends: 100%|██████████| 1/1 [00:00<00:00, 520.26it/s]\n",
      "fitting mode to physics: 100%|██████████| 1/1 [00:00<00:00, 66.88it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cumsum:  [4.13738239e-19 8.27809700e-19 1.24221465e-18 ... 1.00000000e+00\n",
      " 1.00000000e+00 1.00000000e+00]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "binarizing continuous traces filtered fluorescence onphase: 100%|██████████| 1/1 [00:00<00:00, 53.34it/s]\n",
      "binarizing continuous traces filtered fluorescence upphase: 100%|██████████| 1/1 [00:00<00:00, 40.86it/s]\n"
     ]
    }
   ],
   "source": [
    "#######################################\n",
    "#######################################\n",
    "#######################################\n",
    "\n",
    "# binarize the ensembel state    \n",
    "c.plotting = False\n",
    "c.scale_threshold = 0.05 #                 # we want to decrease the burst detection threshold manually - because the \n",
    "                                            # automatic threshold detection is too conservative \n",
    "c.dff_threshold = 0.1\n",
    "c.percentile_threshold = 0.999\n",
    "\n",
    "#\n",
    "#print (\"TODO: need to add the standaline binairzation code to the binarize2pcalcium package....\")\n",
    "#print (\" This is generally a good idea anyways for doing single time sieres extraction\")\n",
    "\n",
    "# right now only binarizing upphase - if want to do negative phase have to invert the signal\n",
    "c.sample_rate = 30\n",
    "c.binarize_ensemble_state_multi()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#######################################################\n",
    "#######################################################\n",
    "#######################################################\n",
    "\n",
    "plt.figure()\n",
    "t = np.arange(c.sessions[1].e_state.shape[0])/30.\n",
    "\n",
    "# make viridis colormap for line plotting up to 7 sessions\n",
    "cmap = matplotlib.cm.get_cmap('viridis')\n",
    "clrs = cmap(np.linspace(0, 1, 7))\n",
    "\n",
    "#\n",
    "for session_id in range(1,8,1):\n",
    "    temp = c.sessions[session_id].e_state_bin \n",
    "    temp2 = c.sessions[session_id].e_state\n",
    "\n",
    "    plt.plot(t, temp + 2*session_id, \n",
    "             label = \"bmi \"+str(session_id),\n",
    "             c='black',\n",
    "             alpha=.5)\n",
    "    \n",
    "    plt.plot(t, temp2 + 2*session_id, \n",
    "             c=clrs[session_id-1])\n",
    "\n",
    "#\n",
    "plt.xlabel(\"Time (sec)\",fontsize=16)\n",
    "\n",
    "#\n",
    "plt.xlim(0, t[-1])\n",
    "plt.yticks([])\n",
    "\n",
    "#c.sessions[1]\n",
    "plt.legend()\n",
    "\n",
    "#\n",
    "plt.suptitle(\"Ensemble state + Binarization\", fontsize=16)\n",
    "\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "a7ba3175",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# of snippets:  292  # of detected snippets of sign:  1   292\n",
      "# of bursts:  292  # of rewarded bursts:  46\n",
      "self.update_spreadsheet:  True\n",
      "TO ADD: grab the tones also during the bursts\n"
     ]
    }
   ],
   "source": [
    "#######################################\n",
    "#######################################\n",
    "#######################################\n",
    "\n",
    "#\n",
    "#c.sessions[session_id].thresh = thresh\n",
    "c.overwrite_spreadsheet = True\n",
    "c.plotting = True\n",
    "c.update_spreadsheet = True\n",
    "c.burst_sign = 1         # -1 or +1\n",
    "c.process_snippets()           # processes snippets and adds them to the spreassheet \n",
    "\n",
    "print (\"TO ADD: grab the tones also during the bursts\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "c2267076",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snippets_warped:  (46, 30)\n"
     ]
    }
   ],
   "source": [
    "###############################################\n",
    "###############################################\n",
    "###############################################\n",
    "\n",
    "#\n",
    "#c.time_warp = False\n",
    "c.plotting = True\n",
    "\n",
    "#c.use_rewarded_bursts_only = False\n",
    "c.get_mean_warped_trace()  # this computes the warp trace regardless"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "92d9e7a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "################################################\n",
    "######### VISUALIZE WARPING FUNCITONS ##########\n",
    "################################################\n",
    "\n",
    "#\n",
    "c.use_stretched = True\n",
    "c.plotting = True\n",
    "c.burst_to_threshold_only = True                 # this uses the bursts only to the threshold\n",
    "c.compute_warping_functions()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "d68772cd",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/354 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 354/354 [00:17<00:00, 19.67it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "burst_raw:  (354, 46, 400)\n",
      "post_burst_raw:  (354, 46, 100)\n",
      "burst_raw_mean:  (354, 400)\n",
      "(46, 200) (46, 30) (46, 100)\n",
      "traces_all:  (354, 46, 330)\n"
     ]
    }
   ],
   "source": [
    "# this plots the best cell responses to bursts\n",
    "c.start_idx = 0\n",
    "c.pre_burst_n_time_pts = 200\n",
    "c.ca_type = 'DFF'\n",
    "\n",
    "# \n",
    "c.plot_warped_cell_ca_trial_and_burst()\n",
    "\n",
    "#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "db098d80",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "n_cells:  263 # rewarsds:  39\n",
      "traces-raw:  (263, 39, 400)\n",
      "traces mean:  (263, 400)\n"
     ]
    }
   ],
   "source": [
    "################################################\n",
    "######### VISUALIZE WARPING FUNCITONS ##########\n",
    "################################################\n",
    "# this plots the best cell responses to bursts\n",
    "c.start_idx = 0\n",
    "c.plot_warped_cell_ca_burst_only()\n",
    "\n",
    "#\n",
    "print (\"SOME TODOS\")\n",
    "print (\" 1. Warp only to the threshold - not the peak of burst\")\n",
    "print (\" 2. compute the rate map for each cell for burst initiation to threshold/end of peak (2 options)\")\n",
    "print (\" 3. compute the rate map for each cell from trial beginning to threshold/end of peak (2 options)\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0f4ae1a6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "da36fbf2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c32fe3f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['bmi/pair_wise_correlations/DON-011733/20230302/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230303/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230304/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230305/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230306/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230307/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230308/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230309/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230329/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230330/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230331/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230402/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230403/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230404/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230405-18threshold/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230406/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230407/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230505/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230509/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230611/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230612/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230613/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230614/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230615/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230616/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230617/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230618/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230619/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230129/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230130/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230131/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230201/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230202/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230204/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230205/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230206/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230330/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230331/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230401/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230402/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230403/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230404/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230406/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230407/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230408/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230504/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230506/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230507evening/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230507morning/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230508/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230510/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230511/plane0/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230302/plane0/backup/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230302/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230303/plane0/backup/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230303/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230304/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230305/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230306/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230307/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230308/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230309/plane0/backup/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-011733/20230309/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230329/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230330/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230331/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230402/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230403/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230404/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230405-18threshold/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230406/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014451/20230407/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230509/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230611/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230612/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230613/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230614/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230615/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230616/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230617/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230618/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-015801/20230619/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230129/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230130/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230131/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230201/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230202/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230204/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230205/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-012502/20230206/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230330/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230331/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230401/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230402/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230403/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230404/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230406/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230407/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014371/20230408/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230504/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230505/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230506/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230507evening/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230507morning/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230508/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230510/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230511/plane0/merged/binarized_traces.npz'\n",
      " 'bmi/pair_wise_correlations/DON-014618/20230504/plane0/merged/old/binarized_traces.npz']\n"
     ]
    }
   ],
   "source": [
    "# load a list of filenames\n",
    "d = np.loadtxt('/home/cat/files.txt',dtype='str')\n",
    "print (d)\n",
    "\n",
    "# convert the array to a dataframe\n",
    "df = pd.DataFrame(d, columns=['filenames'])\n",
    "\n",
    "# add two columns of zeros \n",
    "df['started'] = 0\n",
    "df['completed'] = 0\n",
    "\n",
    "# save the dataframe to an .xlsx file\n",
    "df.to_excel('/home/cat/files.xlsx', index=False)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "138bde45",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6d025ea6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "023ae69b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c181a8e7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86d1cefa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d99d623e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "926f7654",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2b502f6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e99976a9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "382188ed",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e291d36",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.18"
  },
  "toc-autonumbering": false,
  "toc-showcode": true
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
